include/h2o.h

```diff
@@ -503,6 +558,11 @@ struct st_h2o_context_t {
         h2o_timestamp_string_t *value;
     } _timestamp_cache;
 
+    /**
+     * counter for errors internally emitted by h2o
+     */
+    h2o_emitted_errors_t emitted_errors;
```

Comment:
Re logging of status codes.

My understanding is that the intention is not to log the status codes in general, but is to log the errors that originated within H2O.  That means that the source of the error and the logic that updates the counter can be tightly coupled, and IMO we should better couple them instead of passing the number as a value and then using switch-case to determine the slot for two reasons: 1) speed 2) ease of maintenance.

There could be various ways to couple them, but my two cents go to the following approach.

1. rename `h2o_send_error` to `h2o_send_error_generic` (or something)
2. rewrite invocations of `h2o_send_error` taking constant status codes to `h2o_send_errorXXX`, where XXX is the status code.
3. define all the `h2o_send_errorXXX` functions using macro expansion.  The functions should increment the corresponding counters and then call `h2o_send_error_generic`

How does it sound?